src/main/dynamo/Model.ts

```diff
@@ -27,10 +28,36 @@ export class Model<
     } = this
     return new PartitionAndSortKey(
       this.table.partitionKeyName,
-      this.buildKey(partitionKeyPrefix, params[partitionKeyField]),
+      this.buildKey(
+        partitionKeyPrefix,
+        this._buildPartitionKey(params, partitionKeyField)),
 
       this.table.sortKeyName,
-      this.buildKey(sortKeyPrefix, params[sortKeyField]),
+      this.buildKey(
+        sortKeyPrefix,
+        this._buildSortKey(params, sortKeyField)),
+      this.modelTag
+    )
+  }
+
+  partialKey(
+    params: AtLeastOne<{ [X in U]: string }> & AtLeastOne<{ [Y in V]: string }>
```

Comment:
This is where I think I could possibly use a generated type of only valid combinations of keys.